asb-native/src/main/java/org/ballerinax/asb/receiver/MessageReceiver.java

```diff
@@ -513,4 +354,135 @@ public Object closeReceiver() {
             return ASBUtils.returnErrorValue(e.getClass().getSimpleName(), e);
         }
     }
+
+    /**
+     * Converts AMPQ Body value to Java objects.
+     *
+     * @param amqpValue AMQP Value type object
+     */
+    private Object convertAMQPToJava(String messageId, Object amqpValue) {
+        log.debug("Type of amqpValue object  of received message " + messageId + " is " + amqpValue.getClass());
```

Comment:
Better to check if log.isDegungenabled() and then do the debug log as this is called per message?

Reply:
@abeykoon It seems the check won't have an significant impact on the performance the as the `log.debug()` API itself has an early exit checking for the activated debug level